react-carousel-component/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,71 @@
+import { NavButton } from './NavButton';
+import { Indicators } from './Indicators';
+import { ImageBanner } from './ImageBanner';
+import { useEffect, useState } from 'react';
+
+type Props = {
+  images: string[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [timeoutId, setTimeoutId] = useState<number>();
+
+  useEffect(() => {
+    let id = 0;
+    if (timeoutId === undefined) {
+      id = setTimeout(
+        () => setCurrentIndex((prev) => (prev + 1) % images.length),
+        3000
+      );
+    }
+    return () => clearTimeout(id);
+  }, [currentIndex, timeoutId, images.length]);
+
+  function handleIncrementClick(): void {
+    setCurrentIndex((currentIndex + 1) % images.length);
+    timeoutCarousel();
+  }
+
+  function handleDecrementClick(): void {
+    setCurrentIndex((currentIndex - 1 + images.length) % images.length);
+    timeoutCarousel();
+  }
+
+  function handleSelectionClick(index: number): void {
+    setCurrentIndex(index);
+    timeoutCarousel();
+  }
+
+  function timeoutCarousel() {
```

Comment:
I'm not convinced this function is necessary since this work will be done in the `useEffect`, which will run thanks to `currentIndex` being in the dependency array.